crates/sui-types/src/base_types.rs

```diff
@@ -1050,6 +1052,14 @@ impl SequenceNumber {
 
         SequenceNumber(max_input.0 + 1)
     }
+
+    pub fn is_cancelled(&self) -> bool {
+        self == &SequenceNumber::CANCELLED_READ || self == &SequenceNumber::CONGESTED
+    }
+
+    pub fn is_valid(&self) -> bool {
```

Comment:
Doesn't have to be in this PR but I would love to see us explore explicit type wrapping to avoid all the assertions we are doing with is_valid()

Reply:
Will do it in a follow up PR. To make sure I understand, the idea is that for places it expects a valid version, it should guaranteed to get a valid version?